api/transport/router.go

```diff
@@ -37,6 +37,21 @@ type Procedure struct {
 
 	// HandlerSpec specifiying which handler and rpc type.
 	HandlerSpec HandlerSpec
+
+	// Encoding of the handler, for introspection.
+	Encoding Encoding
+
+	// Signature of the handler, for introspection. This should be a snippet of
+	// Go code representing the function definition.
```

Comment:
Not sure what you mean "snippet of Go code" in this context - don't you mean "a string"?